engine/src/main/java/com/datatorrent/stram/debug/TupleRecorder.java

```diff
@@ -76,6 +76,31 @@
   private String recordingNameTopic;
   private long numWindows = Long.MAX_VALUE; // number of windows to record
   private Runnable stopProcedure; // stop procedure to execute
+
+  private static final Logger logger = LoggerFactory.getLogger(TupleRecorder.class);
+
+  // If there are errors processing tuples, don't log an error for every tuple as it could overwhelm the logs.
+  // The property specifies the minumum number of tuples between two consecutive error log statements. Set it to zero to
+  // log every tuple error
+  private static int ERROR_LOG_GAP;
+  int lastLog = -1;
+
+  static {
+    int gap = 10000;
+    String property = System.getProperty("apex.tupleRecorder.errorLogGap");
+    if (property != null) {
+      try {
+        gap = Integer.parseInt(property);
+        if (gap < 0 ) {
+          throw new IllegalArgumentException("Error log gap should be greater than or equal to 0");
```

Comment:
This may lead to containers termination due to RuntimeException in the middle of a run when tuple recording is enabled. It should be sufficient to log value of the settings and in case of incorrectly set property, set it to the default.